python/desc/slrealizer/plotting.py

```diff
@@ -37,57 +34,92 @@ def determineScale(lensX, sourceX, lensY, sourceY):
 
 def plotFigureOnMatplotlib(currObs, convolve, quasar_alpha, lens_alpha, sourceX, sourceY, lensX, lensY, plotX, plotY):
     #fig = plt.figure(1)
-    scale_factor = 2
-    if(currObs[1]=='g'):
-        circleColor = 'g'
+    if (currObs[1]=='u'):
+        circleColor = 'violet'
+    elif (currObs[1]=='g'):
+        circleColor = 'blue'
+    elif (currObs[1]=='r'):
+        circleColor = 'green'
+    elif (currObs[1]=='i'):
+        circleColor = 'orange'
+    elif (currObs[1]=='z'):
+        circleColor = 'red'
     else:
-        circleColor = 'r'
+        raise ValueError('Unknown filter name '+currObs[1])
+
     plt.axis('scaled')
     plt.ylim(-3, 3)
     plt.xlim(-3, 3)
     plt.xlabel('xPosition')
     plt.ylabel('yPosition')
     plt.title('Observation with ' + 'filter ' + currObs[1] + ' on MJD ' + str(currObs[0]))
+
+    scale_factor = 2
+    PSF_HWHM = currObs[2]/scale_factor
+
+    # Draw quasar images:
     for i in range(4):
-        source = plt.Circle((sourceX, sourceY), radius=currObs[2]/scale_factor, alpha=quasar_alpha, fc=circleColor, linewidth=0)
-        if(convolve):
-            # For now, we just let lensFWHM to be zero.
-            #lensFWHM = currLens['REFF']
-            lensFWHM = 0.0
-            lens = plt.Circle((lensX[i]+sourceX, lensY[i]+sourceY), radius=convolve(currObs[2], lensFWHM)/scale_factor, alpha=lens_alpha, fc=circleColor, linewidth=0)
-        else:
-            lens = plt.Circle((lensX[i]+sourceX, lensY[i]+sourceY), radius=currObs[2]/scale_factor, alpha=lens_alpha, fc=circleColor, linewidth=0, edgecolor='b', linestyle='dashed')
-        plt.gca().add_patch(lens)
-        plt.gca().add_patch(source)
+        source = plt.Circle((sourceX, sourceY),
+                            radius=PSF_HWHM,
+                            alpha=quasar_alpha,
+                            fc=circleColor, linewidth=0)
+    # Draw lens galaxy:
+    if(convolve):
+        # For now, we just let lensFWHM to be zero.
+        #lensFWHM = currLens['REFF']
+        lensFWHM = 0.0
+        galaxyHWHM = convolve(currObs[2], lensFWHM)/scale_factor
+        # Bug: the LENS position is in the CENTER of the field.
+        #      The SOURCE position is UNOBSERVABLE.
+        #      The IMAGE positions are in teh OM10 catalog in the
+        #      XIMG and YIMG columns.
+        lens = plt.Circle((lensX[i]+sourceX, lensY[i]+sourceY),
```

Comment:
I re-organized this section of the code according to your varaible names - see how my comments match them? I'll issue you with fixing the variable names and getting the images and magnifications right.